Tests/SwiftParserTest/translated/StringLiteralEofTests.swift

```diff
@@ -123,18 +130,14 @@ final class StringLiteralEofTests: XCTestCase {
     AssertParse(
       ##"""
       _ = """
-          \(1️⃣"bar2️⃣
-      3️⃣    baz4️⃣
+          \("1️⃣bar
+      2️⃣    baz3️⃣
       """##,
       diagnostics: [
-        // TODO: Old parser expected error on line 1: unterminated string literal
-        DiagnosticSpec(message: "expected value in string literal"),
-        DiagnosticSpec(message: #"unexpected code '"bar' in string literal"#),
-        DiagnosticSpec(locationMarker: "2️⃣", message: #"expected ')' in string literal"#),
-        DiagnosticSpec(locationMarker: "3️⃣", message: #"unexpected code '' in string literal"#),
-        DiagnosticSpec(locationMarker: "4️⃣", message: #"expected '"""' to end string literal"#),
-        // TODO: Old parser expected error on line 3: cannot find ')' to match opening '(' in string interpolation
-        // TODO: Old parser expected error on line 3: unterminated string literal
+        DiagnosticSpec(locationMarker: "1️⃣", message: #"expected '"' to end string literal"#),
+        DiagnosticSpec(locationMarker: "1️⃣", message: "unexpected code in string literal"),
+        DiagnosticSpec(locationMarker: "2️⃣", message: "expected ')' in string literal"),
+        DiagnosticSpec(locationMarker: "3️⃣", message: #"expected '"""' to end string literal"#),
```

Comment:
Bit of a weird case I suppose, but it's a little strange that we don't treat this as `bar` being part of the string within the interpolation. Though having a string within an interpolation seems fairly unusual anyway, so... maybe doesn't actually matter.

I think ideally we'd have just two diagnostics: one for the missing `")` (or two, whatever) and another for the missing `"""`.